### PR TITLE
Refactor Dataset Pipeline

### PR DESCRIPTION
**Pull request by amarouane-ABDELHAK**
**Summary:** Summary of changes

Refactoring the DAGs so we will reuse the tasks

## Changes

* Moved build stack to processing group 
* Refactor dataset-pipeline DAG

### CODE REVIEW

**Comment by amarouane-ABDELHAK** [comment]
Tested this locally with 
collection definition
```json
{
    "assets": {
        "ABI L1b Radiances": {
            "description": "Single emissive band ABI L1b Radiance Products are digital maps of outgoing radiance values at the top of the atmosphere for IR bands",
            "title": "ABI L1b Radiances"
        },
        "thumbnail": {
            "description": "Photo by [NOAA](https://eoimages.gsfc.nasa.gov/images/imagerecords/151000/151897/mississippi_oli_2023259_lrg.jpg) (Landsat 8 — OLI image of the Mississippi River below Memphis, Tennessee on September 16, 2023 at near record low water levels, limiting barge shipments, threatening drinking water supplies, agriculture, and ecosystems.)",
            "href": "https://thumbnails.openveda.cloud/goes-16-cover.png",
            "roles": [
                "thumbnail"
            ],
            "title": "Thumbnail",
            "type": "image/jpeg"
        }
    },
    "collection": "goes-16-l1b-sid",
    "description": "Single emissive band ABI L1b Radiance Products are digital maps of outgoing radiance values at the top of the atmosphere for IR bands.",
    "discovery_items": [
        {
            "bucket": "veda-data-store-dev",
            "cogify": false,
            "discovery": "s3",
            "dry_run": false,
            "filename_regex": ".*.tif$",
            "id_template": "{}",
            "prefix": "cyclone/GOES/"
        }
    ],
    "extent": {
        "spatial": {
            "bbox": [
                [
                    -151.8686894284635,
                    14.560590614318777,
                    -52.90605171021198,
                    56.772841751410404
                ]
            ]
        },
        "temporal": {
            "interval": [
                [
                    "2024-06-08 00:00:00+00",
                    "2024-06-08 00:00:00+00"
                ]
            ]
        }
    },
    "is_periodic": true,
    "item_assets": {
        "ABI L1b Radiances": {
            "description": "Single emissive band ABI L1b Radiance Products are digital maps of outgoing radiance values at the top of the atmosphere for IR bands",
            "roles": [
                "data",
                "layer"
            ],
            "title": "ABI L1b Radiances",
            "type": "image/tiff; application=geotiff; profile=cloud-optimized"
        }
    },
    "license": "MIT",
    "links": [
        {
            "href": "https://dev.openveda.cloud/api/stac/collections/goes-16-l1b/items",
            "rel": "items",
            "type": "application/geo+json"
        },
        {
            "href": "https://dev.openveda.cloud/api/stac/",
            "rel": "parent",
            "type": "application/json"
        },
        {
            "href": "https://dev.openveda.cloud/api/stac/",
            "rel": "root",
            "type": "application/json"
        },
        {
            "href": "https://dev.openveda.cloud/api/stac/collections/goes-16-l1b",
            "rel": "self",
            "type": "application/json"
        }
    ],
    "providers": [
        {
            "name": "NOAA",
            "roles": [
                "producer",
                "processor",
                "licensor"
            ],
            "url": "https://noaa-goes16.s3.amazonaws.com/index.html"
        },
        {
            "name": "NASA VEDA",
            "roles": [
                "host"
            ],
            "url": "https://www.earthdata.nasa.gov/dashboard/"
        }
    ],
    "renders": {
        "dashboard": {
            "assets": [
                "ABI L1b Radiances"
            ],
            "colormap_name": "greys_r",
            "rescale": [
                [
                    0,
                    5
                ]
            ]
        }
    },
    "stac_extensions": [
        "https://stac-extensions.github.io/render/v1.0.0/schema.json",
        "https://stac-extensions.github.io/item-assets/v1.0.0/schema.json"
    ],
    "stac_version": "1.0.0",
    "summaries": {
        "datetime": [
            "2024-06-08T00:00:00Z",
            "2024-06-08T00:00:00Z"
        ]
    },
    "time_density": "daily",
    "title": "GOES-16 ABI L1b Radiances",
    "type": "Collection"
}
```
runs successfully 
https://dev.openveda.cloud/api/stac/collections/goes-16-l1b-abdelhak
https://dev.openveda.cloud/api/stac/collections/goes-16-l1b-abdelhak/items
image

File: dags/veda_data_pipeline/groups/discover_group.py [hunk 1]
```diff
@@ -56,6 +49,36 @@ def discover_from_s3_task(ti=None, event={}, alt_payload = None, **kwargs):
...
+            results.append({
+                "run_id": f"{dag_run_id}_{uuid.uuid4()}_{indx}",
+                **base_payload,
+                "payload": payload_xcom,
+            })
+
+    return results
+
+@task
+@deprecated(reason="Please use get_files_task function that hundles both files and dataset files use cases")
```
Comment (botanical) [inline dags/veda_data_pipeline/groups/discover_group.py]:
```suggestion
@deprecated(reason="Please use get_files_task function that handles both files and dataset files use cases")
```

File: dags/veda_data_pipeline/groups/processing_tasks.py [hunk 2]
```diff
@@ -36,5 +36,9 @@ def submit_to_stac_ingestor_task(built_stac: dict):
     return event
 
 
-
-
+@task(max_active_tis_per_dag=5)
+def build_stac_task(payload):
+    from veda_data_pipeline.utils.build_stac.handler import stac_handler
```
Comment (botanical) [inline dags/veda_data_pipeline/groups/processing_tasks.py:41]: Any reason to not import this at the top of the file? 😯

Comment (amarouane-ABDELHAK) [reply dags/veda_data_pipeline/groups/processing_tasks.py:41]: When the scheduler reads the DAGs and the tasks to be scheduled, it executes the imports if they are at the top of the script. To keep the scheduler lightweight, we avoid installing non-Airflow libraries on it. Instead, we import the required libraries at the task level, where the worker executes the function. This approach is explained [here](https://github.com/NASA-IMPACT/veda-data-airflow/blob/dev/dags/veda_data_pipeline/utils/s3_discovery.py#L71C16-L71C30).

**Review by anayeaye — APPROVED** [review]

**Pull request merged**